common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -403,22 +392,30 @@ public boolean close(T trackedObject) {
 
         @Override
         public String toString() {
-            if (creationRecord == null) {
+            if (head == null) {
                 return EMPTY_STRING;
             }
 
-            final Object[] array;
-            final int removedRecords;
-            if (lastRecords != null) {
-                synchronized (lastRecords) {
-                    array = lastRecords.toArray();
-                    removedRecords = this.removedRecords;
+            final String creationRecord = head.toString();
+            final String[] array;
+            int idx = 0;
+            String last = null;
+
+            synchronized (head) {
+                array = new String[numRecords];
+                Record record = head.next;
+                while (record != null) {
+                    String recordStr = record.toString();
+                    if (last == null || !last.equals(recordStr)) {
```

Comment:
I just copied what was there before. I would need to check in more detail but from a quick glance it looks ok.

Reply:
oh I see ... this was done previously at the record time, but I guess you are delaying the check bcz you may be able to avoid it all together unless there is a leak ... lgtm!

Reply:
exactly.. Mostly because toString() is expensive here as it calls getStackTrace()